flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -535,7 +541,9 @@ private void updateStatistics(BufferConsumer buffer) {
     }
 
     private void updateStatistics(Buffer buffer) {
-        totalNumberOfBytes += buffer.getSize();
+        if (buffer.isBuffer()) {
+            totalNumberOfBytes += buffer.getSize();
```

Comment:
NB! Right now, The totalNumberOfBytes calculate only the size of the data buffers(without event) which broke the old behavior but as I can see we use it only in toString. So maybe it is not a problem but if it is, we can add another variable for data buffers size calculation.

Reply:
But why are you changing this in the first place? What's your motivation?

Reply:
scratch that. Now I can see why have you change this. I think this change would be fine, assuming we want the `outputQueueSize` to ignore events. If `outputQueueLength` is ignoring events, then I would vote for `outputQueueSize` to ignore them as well. In other words, they should stay in sync.

Reply:
`outputQueueLength` takes into account all buffers including the event. So I implemented the same logic for `outputQueueSize`